File: backend/dashboard/static/dashboard/js/views/drives.js
Dash.render(async (root) => {
  const { api, escapeHTML, pct, fmtRel } = Dash;
  const d = await api("/dashboard/api/drives");
  if (!d) return (root.innerHTML = `<div class="empty">Indisponible.</div>`);
  const LABEL = {curiosity:"Curiosité",social:"Social",expression:"Expression",rest:"Repos"};

  root.innerHTML = `
    <div class="grid cols-3 mb">
      <div class="card">
        <h3>Énergie globale</h3>
        <div class="stat-value">${pct(d.energy)}</div>
        <div class="bar mt ok"><div class="fill" style="width:${d.energy*100}%"></div></div>
      </div>
      <div class="card">
        <h3>Drive dominant</h3>
        <div class="stat-value" style="color:var(--magenta)">${d.dominant || "—"}</div>
        <div class="stat-sub">contribution conscience : ${d.conscience_contribution}</div>
      </div>
      <div class="card">
        <h3>Détail conscience</h3>
        <div class="mono muted" style="word-break:break-all">${escapeHTML(d.conscience_label || "—")}</div>
      </div>
    </div>

    <div class="grid cols-2">
      ${d.drives.map(dr => `
        <div class="card">
          <h3>${LABEL[dr.kind] || dr.kind}<span class="tag">${dr.kind === "rest" ? "fatigue" : "motivation"}</span></h3>
          <div class="flex between center mb">
            <span class="stat-value" style="font-size:22px">${pct(dr.tension)}</span>
            <span class="muted">tension</span>
          </div>
          <div class="bar ${dr.kind==='rest'?'warn':''} mb"><div class="fill" style="width:${dr.tension*100}%"></div></div>
          <div class="metric-row"><span class="k">Seuil satisfaction</span><span class="v">${pct(dr.params.satisfy_threshold)}</span></div>
          <div class="metric-row"><span class="k">Poids conscience</span><span class="v">${dr.params.weight.toFixed(2)}</span></div>
          <div class="metric-row"><span class="k">Croissance</span><span class="v">${dr.params.growth_rate}/s</span></div>
          <div class="metric-row"><span class="k">Décroissance à satisf.</span><span class="v">${pct(dr.params.decay_on_satisfy)}</span></div>
          <div class="metric-row"><span class="k">Dernière satisfaction</span><span class="v">${fmtRel(new Date(dr.last_satisfied*1000).toISOString())} ago</span></div>
        </div>`).join("")}
    </div>`;
});

File: backend/dashboard/views/api/drives.py
from __future__ import annotations

from django.http import JsonResponse
from django.views.decorators.http import require_http_methods


@require_http_methods(["GET"])
def drives(request):
    from drives.engine import drive_engine
    from drives.state import DEFAULT_PARAMS

    drive_engine.update()
    rows = []
    for kind, state in drive_engine.states.items():
        params = DEFAULT_PARAMS[kind]
        rows.append({
            "kind": kind.value,
            "tension": round(state.tension, 3),
            "last_update": state.last_update,
            "last_satisfied": state.last_satisfied,
            "params": {
                "growth_rate": params.growth_rate,
                "decay_on_satisfy": params.decay_on_satisfy,
                "weight": params.weight,
                "satisfy_threshold": params.satisfy_threshold,
            },
        })
    dominant = drive_engine.get_dominant()
    bonus, label = drive_engine.conscience_contribution()
    return JsonResponse({
        "drives": rows,
        "dominant": dominant.kind.value if dominant else None,
        "energy": round(drive_engine.energy_level(), 3),
        "conscience_contribution": round(bonus, 3),
        "conscience_label": label,
    })
